### ISSUE TITLE
Webview does not dispose properly

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#50556]
I am testing from the `ben/editor` branch that contains the changes for Grid. If you split an editor to the side and then close that editor, we now actually dispose the editor and no longer keeping it around as we used to before. This means that we now rely on the editor freeing up its resources properly.

For any webview editor (e.g. markdown preview or extensions editor) I see lots of controls still hanging around after:
* open a single of these editors
* split it once
* close the second instance
* create a heap snapshot
* repeat split/close a couple of times
* create a heap snapshot again
* compare it to the first

![image](https://user-images.githubusercontent.com/900690/40606208-f6460fc6-6264-11e8-8b4c-58f0f94e4cbf.png)

**Comment by mjbvz** [comment]
@bpasero I've fixed some of the leaks on our side but the leaks still exist. Can you take a this possible electron issue to see if you have any thoughts on the problem: https://github.com/electron/electron/issues/13243

**Comment by bpasero** [comment]
@mjbvz did you test with our previous Electron 1.7.12 to see if this is a regression from the 2.0 update?

**Comment by mjbvz** [comment]
Not a regression from 1.7. I've minimized the impact of this problem. Certainly possible I'm doing something wrong in the code here

**Comment by ivankravets** [comment]
I have a similar issue and I don't know should I open a new one. 

# Code

```js
panel = vscode.window.createWebviewPanel(...);
this.subscriptions.push(panel.onDidDispose(this.onPanelDisposed.bind(this)));

  onPanelDisposed() {
    console.log('onPanelDisposed');
    this._currentPanel = undefined;
  }

  dispose() {
    console.log('Home::dispose', this._currentPanel);
    if (this._currentPanel) {
      console.log('Going to kill home');
      this._currentPanel.dispose();
      this._currentPanel = undefined;
    }
}
```

# Problem

1. I don't know is it a feature or bug of VSCode, but each time when we add a new folder to workspace VSCode kills our extension via `deactivate()` and then `activate()` again. This workflow requires a lot of CPU resources for our `activate()` implementation. https://github.com/platformio/platformio-vscode-ide/blob/develop/src/main.js

2. When VSCode calls `deactivate()`, we call `panel.dispose()`, to close active PIO Home tab. Sometimes it works, sometimes does not. See log from debug console:
```
deactivating extension
extension.js:409
Home::dispose
home.js:132
Object
_isDisposed:false
_visible:true
_active:true
_onDisposeEmitter:Object
_onDidChangeViewStateEmitter:Object
_handle:"1"
_proxy:Object
_viewType:"pioHome"
_options:Object
_viewColumn:1
_title:"PIO Home"
_webview:Object
_iconPath:Object
Going to kill home
home.js:134
onPanelDisposed
home.js:127
activating extension
extension.js:26
starting language server
```

# Result

Our panel is not closed

**Comment by mjbvz** [comment]
@ivankravets Unrelated to original problem. Please open a new issue

**Comment by mjbvz** [comment]
I investigated this again and made a few more fixes but am still blocked by the upstream electron issue. 

The best long term fix here is probably to move off of webviews and to normal iframes instead

**Comment by ivankravets** [comment]
How to use native iframes? Do you have example?

**Comment by mjbvz** [comment]
Tested using iframe based webviews using the following steps:

1. Set `"webview.experimental.useIframes": true` to force iframes to be used
2. Open a markdown file. Open the preview once and then close it
3. Snapshot
4. Now, multiple times: open the preview and then close it
5. With the preview closed, take another snapshot (give GC a few seconds to run).

You can now compare between the snapshots. Search for `webview` and confirm that no webview elements are still around.  Make sure you are looking at the `objects allocated between snapshot X and Y` since there will still be some static webview objects around (such as the services):

![Screen Shot 2021-05-05 at 5 21 48 PM](https://user-images.githubusercontent.com/12821956/117226083-b06e1180-adc8-11eb-9f52-98980cb69bd0.png)
